setup.py

```diff
@@ -122,32 +174,31 @@ def build_extension(self, ext):
         if not has_ninja() or self.parallel:
             build_args += ["-j{}".format(self.parallel) if self.parallel else "-j"]
 
-        cmake_args += ["-DBUILD_GUI_VIEWERS={}".format("ON" if not HEADLESS else "OFF")]
-        cmake_args += ["-DBUILD_TESTS={}".format("ON" if BUILD_TESTS else "OFF")]
+        cmake_args += [
+            "-DBUILD_GUI_VIEWERS={}".format("ON" if not args.headless else "OFF")
+        ]
+        cmake_args += ["-DBUILD_TESTS={}".format("ON" if args.build_tests else "OFF")]
 
         env = os.environ.copy()
         env["CXXFLAGS"] = '{} -DVERSION_INFO=\\"{}\\"'.format(
             env.get("CXXFLAGS", ""), self.distribution.get_version()
         )
 
-        if not os.path.exists(self.build_temp):
-            os.makedirs(self.build_temp)
-
         if self.run_cmake(cmake_args):
             subprocess.check_call(
-                ["cmake", ext.sourcedir] + cmake_args, cwd=self.build_temp, env=env
+                shlex.split("cmake -H{} -B{}".format(ext.sourcedir, self.build_temp))
+                + cmake_args,
+                env=env,
             )
 
-        subprocess.check_call(
-            ["cmake", "--build", "."] + build_args, cwd=self.build_temp
-        )
+        subprocess.check_call(shlex.split("cmake --build {}".format(self.build_temp)))
```

Comment:
This is missing the `build_args`, so `--config` or `-jN` doesn't get set correctly. This worked for me:

```suggestion
        subprocess.check_call(shlex.split("cmake --build {}".format(self.build_temp)) + build_args)
```

Reply:
Whoops!